noxfile.py

```diff
@@ -117,8 +119,14 @@ def set_dev(session):
     """Install PyBaMM in editable mode."""
     set_environment_variables(PYBAMM_ENV, session=session)
     envbindir = session.bin
-    session.install("-e", ".[all]", silent=False)
+    session.install("virtualenv")
+    session.run("virtualenv", os.fsdecode(VENV_DIR), silent=True)
+    python = os.fsdecode(VENV_DIR.joinpath("bin/python"))
+    session.run(python, "-m", "pip", "install", "-e", ".[all]", silent=False)
     session.install("cmake", silent=False)
+    session.install("-e", ".[dev]", external=True)
```

Comment:
We should install `cmake` before we install the other dependencies, otherwise PyBaMM will build itself without its compiled components. Also, the extra dependencies can be installed at once (i.e., `[all,dev]`) to not build the wheels more than once